### PR TITLE
Update libimobiledevice packages, fix usbmuxd

### PR DESCRIPTION
**Pull request by infinisil**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/54953 updated usbmuxd only, seemingly without testing, because when I updated my system today it didn't work anymore, the issue being https://github.com/libimobiledevice/usbmuxd/issues/95. All these libimobiledevice packages should be updated together such that such things don't happen. This PR updates all of them to their latest master version. I also changed ifuse to use the master version instead of its releases, because there hasn't been a release in a while (see https://github.com/libimobiledevice/ifuse/issues/34).

Ping @worldofpeace @arilotter 

This is currently also broken for 19.03 too, <s>PR for a partial backport is incoming</s> after discussions with @grahamc and @samueldr on IRC I decided to instead backport all of these updates, because libimobiledevice doesn't have any stable versions anyways, and not updating all those packages roughly at the same time might lead to more such issues.

###### Things done

I have verified that this fixes above issue, allowing me to use my iPhone for tethering again. I also checked that ifuse can mount the filesystem and that ideviceinstaller can list installed apps.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
Weird, I tested that on a deployed system :confused:

**Comment by worldofpeace** [comment]
Testing this now on my machine @Infinisil

**Review by worldofpeace — APPROVED** [review]
Things tested with IOS 12.2

* `usbmuxd` works
  No problems when I inspect the output in the journal
* `ifuse` works
* listed applications with `ideviceinstaller -l`

**Comment by worldofpeace** [comment]
And I was using `usbmuxd` with the module btw.

**Comment by Infinisil** [comment]
Ah sorry for the accusation then, I didn't verify that that PR actually broke it. I'm bisecting the issue now to find the cause.

**Comment by worldofpeace** [comment]
> Ah sorry for the accusation then, I didn't verify that that PR actually broke it. I'm bisecting the issue now to find the cause.

That may not be needed because it could have been an issue for certain devices.

**Comment by Infinisil** [comment]
Wait no, I *did* bisect it, and it lead me to https://github.com/nixos/nixpkgs/commit/4bf34f16d8f4e703d4975f854d7c69378123cd8f from #54953

I've been testing `usbmuxd` with just
```
$ nix-build -A usbmuxd
$ sudo result/bin/usbmuxd -U usbmux -f
```

and an iOS device plugged in. In https://github.com/NixOS/nixpkgs/commit/4bf34f16d8f4e703d4975f854d7c69378123cd8f this leads to an immediate failure:

```
[01:53:26.446][3] usbmuxd v1.1.1 starting up
[01:53:26.447][3] Successfully dropped privileges to 'usbmux'
[01:53:26.456][3] Initialization complete
[01:53:26.457][3] Connecting to new device on location 0x3000b as ID 1
[01:53:26.457][3] Connected to v2.0 device 1 on location 0x3000b with serial number 073e1cde3e4009384fc8455c6f95a96456ebf6b0
[01:53:26.458][2] Attempted to connect to nonexistent device 0
[01:53:26.458][1] preflight_worker_handle_device_add: ERROR: Could not connect to lockdownd on device 073e1cde3e4009384fc8455c6f95a96456ebf6b0, lockdown error -8
```

But on 0faa6bab501 (master before the merge):

```
[01:54:14.138][3] usbmuxd v1.1.1 starting up
[01:54:14.139][3] Successfully dropped privileges to 'usbmux'
[01:54:14.149][3] Initialization complete
[01:54:14.150][3] Connecting to new device on location 0x3000b as ID 1
[01:54:14.150][3] Connected to v2.0 device 1 on location 0x3000b with serial number 073e1cde3e4009384fc8455c6f95a96456ebf6b0
```

So it might be that you just didn't check for this error or just didn't test this usecase. Or it might be that something in your stack is different, I'm running a pretty old iOS version, maybe the error regarding `lockdownd` would only occur in older versions.

**Comment by worldofpeace** [comment]
@Infinisil did you want another tester? Otherwise LGTM I'll merge.

**Comment by Infinisil** [comment]
Feel free to merge

**Pull request merged**
